multy_test/utility.cpp

```diff
@@ -42,6 +43,83 @@ size_t dummy_fill_entropy(void*, size_t size, void* dest)
 namespace test_utility
 {
 
+const uint64_t SATOSHIS_IN_BTC = 100L*1000*1000;
+const uint64_t WEIS_IN_ETH = 1000L*1000*1000*1000*1000*1000;
+const uint64_t WEIS_IN_GWEI = 1000L*1000*1000;
+
+
+// Returns Satoshi corresponding to fractional amount of BTC.
+BigInt operator"" _BTC(const long double btc)
+{
+    const double satoshi = btc * SATOSHIS_IN_BTC;
+    if (satoshi < 0)
+    {
+        if(satoshi < std::numeric_limits<int64_t>::min())
+        {
+            THROW_EXCEPTION("Value is too low.");
+        }
+
+        return BigInt(static_cast<int64_t>(satoshi));
+    }
+
+    if(satoshi > std::numeric_limits<uint64_t>::max())
+    {
+        THROW_EXCEPTION("Value is too high.");
+    }
+
+    return BigInt(static_cast<uint64_t>(satoshi));
+}
+
+BigInt operator"" _SATOSHI(const unsigned long long int satoshi)
+{
+    return BigInt(static_cast<uint64_t>(satoshi));
+}
+
+BigInt operator "" _ETH(const long double eth)
+{
+    const double wei = eth * WEIS_IN_ETH;
+    if (wei < 0)
```

Comment:
You should check against max/min values of Ether, which are out of range of int64.

Reply:
Ok, nevermind